Assets/Scripts/Gameplay/ConnectionManagement/ConnectionManager.cs

```diff
@@ -0,0 +1,171 @@
+using System;
+using System.Threading.Tasks;
```

Comment:
We should have tests for this. TestHelpers should help here.
cc @fernando-cortez